features/retry.feature

```diff
@@ -0,0 +1,103 @@
+Feature: Retry flaky tests
+
+  Using the `--retry` flag will retry failing tests for the specified number of times
+  Additionally using the `--retryTagFilter` flag will re-run only tests matching the tag expression
+
+  @spawn
+  Scenario: running Cucumber JS with --retryTagFilter but no positive --retry will fail
+    When I run cucumber-js with `--retryTagFilter @flaky`
+    Then it fails  
+
+  Scenario: retrying a flaky test will eventually make it pass
+    Given a file named "features/a.feature" with:
+      """
+      Feature:
+        Scenario: Flaky
+          Given a flaky step
+      """
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {Given} from 'cucumber'
+
+      let willPass = false
+
+      Given(/^a flaky step$/, function() {
+        if (willPass) {
+          return
+        }
+        willPass = true
+        throw 'fail'        
+      })
+      """
+    When I run cucumber-js with `--retry 1`
+    Then the step "a flaky step" has status "passed"
```

Comment:
I think we should test that this step / scenario ran twice one failing, one passing. Same for below test that we ran two scenarios and both failed.